src/domain/usecases/metadata/GetSharingSummaryUseCase.ts

```diff
@@ -0,0 +1,99 @@
+import _ from "lodash";
+import { FutureData } from "../../entities/Future";
+import { MetadataRepository } from "../../repositories/MetadataRepository";
+import { MetadataPayload } from "../../entities/MetadataItem";
+import { SharingWarning, SharingSummary, SharingPayload } from "../../entities/SharingSummary";
+
+export class GetSharingSummaryUseCase {
+    constructor(private metadataRepository: MetadataRepository) {}
+
+    public execute(ids: string[]): FutureData<SharingSummary> {
+        return this.metadataRepository.getMetadataWithChildren(ids).map(payloads => {
+            const metadataSharingsWithChildren = this.getMetadataSharingWithChildren(payloads, ids);
+            const sharingWarnings = this.cleanMetadataSharing(metadataSharingsWithChildren);
```

Comment:
I think it's not a good name for the function, here you are getting the metadata sharing whose children have different sharing settings than their parents.